plugins/modules/backup_plan_info.py

```diff
@@ -110,10 +110,21 @@
 from ansible_collections.amazon.aws.plugins.module_utils.retries import AWSRetry
 
 
+def get_all_backup_plans_info(client):
+    paginator = client.get_paginator("list_backup_plans")
+    return paginator.paginate().build_full_result()
+
+
 def get_backup_plan_detail(client, module):
     backup_plan_list = []
     backup_plan_names = module.params.get("backup_plan_names")
 
+    if backup_plan_names is None:
+        backup_plan_names = []
+        backup_plan_list_info = get_all_backup_plans_info(client)["BackupPlansList"]
+        for mylist in backup_plan_list_info:
```

Comment:
Could we use a variable name specific to the module rather than a generic or unrelated name like 'mylist'?